analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/CalculationsShouldNotOverflow.cs

```diff
@@ -0,0 +1,183 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    public class Sample
+    {
+        public void Types()
+        {
+            int i = int.MaxValue;
+            _ = i + 1;              // FIXME Non-compliant
+            i = int.MinValue;
+            _ = i - 1;              // FIXME Non-compliant
+
+            uint ui = uint.MaxValue;
+            _ = ui + 1;             // FIXME Non-compliant
+            ui = uint.MinValue;
+            _ = ui - 1;             // FIXME Non-compliant
+
+            short i16 = short.MaxValue;
+            _ = i16 + 1;            // FIXME Non-compliant
+            i16 = short.MinValue;
+            _ = i16 - 1;            // FIXME Non-compliant
+
+            ushort ui16 = ushort.MaxValue;
+            _ = ui16 + 1;           // FIXME Non-compliant
+            ui16 = ushort.MinValue;
+            _ = ui16 - 1;           // FIXME Non-compliant
+
+            long i64 = long.MaxValue;
+            _ = i64 + 1;            // FIXME Non-compliant
+            i64 = long.MinValue;
+            _ = i64 - 1;            // FIXME Non-compliant
+
+            ulong ui64 = ulong.MaxValue;
+            _ = ui64 + 1;           // FIXME Non-compliant
+            ui64 = ulong.MinValue;
+            _ = ui64 - 1;           // FIXME Non-compliant
+
+            int b = byte.MaxValue;
+            _ = b + 1;              // FIXME Non-compliant
+            b = byte.MinValue;
+            _ = b - 1;              // FIXME Non-compliant
+
+            sbyte sb = sbyte.MaxValue;
+            _ = sb + 1;             // FIXME Non-compliant
+            sb = sbyte.MinValue;
+            _ = sb - 1;             // FIXME Non-compliant
+        }
+
+        public void BasicOperators()
```

Comment:
also / (that would be multiplied by something big later), to document the behavior